### PR TITLE
Fix: Console launch fails when Windows username contains spaces

### LINKED ISSUE #11259: can not launch console while windows username have space
### Environment

- Git Extensions 4.2.0.17539
- Build d3b383c77a3cda85b835c7e6325c0ebb146a8be9
- Git 2.42.0.windows.2
- Microsoft Windows NT 10.0.22621.0
- .NET 6.0.23
- DPI 96dpi (no scaling)
- Portable: False
- Microsoft.WindowsDesktop.App Versions

```
    Microsoft.WindowsDesktop.App 6.0.21 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 6.0.23 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 7.0.10 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 7.0.12 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 8.0.0-rc.1.23420.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

### Issue description

can not launch console while windows username have space

### Steps to reproduce

user name :Acker Liu

operable program or batch file.'C:lUsers\Acker'is not recognized nternal external command

image

### Did this work in previous version of GitExtensions?

_No response_

### Diagnostics

_No response_

### PR DESCRIPTION
**Pull request by apoorvdarshan**
## Summary

- Add null/empty guards in `PathUtil.TryFindShellPath` to prevent `Path.Combine` from throwing or producing invalid paths when `ProgramW6432` env var is null or `AppSettings.LinuxToolsDir` is empty (both common when the Windows username contains spaces, e.g. "Acker Liu")
- Add unit tests for `TryFindShellPath` null-safety and `Quote`/`QuoteNE` with space-containing paths
- Add shell command quoting tests verifying `GetChangeDirCommand` and `ExecutableCommandLine` patterns produce correctly quoted output for paths with spaces

Fixes #11259

## Test plan

- [ ] Run `dotnet test --filter PathUtilTest` — new tests pass, existing tests unaffected
- [ ] Run `dotnet test --filter ShellTests` — new shell quoting tests pass
- [ ] Manually verify console launch with a Windows user account containing spaces in the username

### CODE REVIEW

**Review by RussKie — APPROVED** [review]

File: tests/app/UnitTests/GitUI.Tests/Shells/ShellTests.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
+using FluentAssertions;
```
Comment (mstv) [inline tests/app/UnitTests/GitUI.Tests/Shells/ShellTests.cs]: New files should be added with the UTF8 BOM.

**Review by mstv — COMMENTED** [review]
LGTM

Could you add your sign-off to `contributors.txt`, please?

File: tests/app/UnitTests/GitUI.Tests/Shells/ShellTests.cs (hunk 1 above)
Comment (apoorvdarshan) [reply tests/app/UnitTests/GitUI.Tests/Shells/ShellTests.cs]: Done — added UTF-8 BOM to ShellTests.cs in 2fb208c.

**Comment by apoorvdarshan** [comment]
Thanks @mstv\! Added the UTF-8 BOM to ShellTests.cs and signed contributors.txt in 2fb208c.
